test/Npgsql.Tests/Replication/PgOutputReplicationTests.cs

```diff
@@ -0,0 +1,537 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Npgsql.Replication.Logical;
+using Npgsql.Replication.Logical.Protocol;
+using NpgsqlTypes;
+using NUnit.Framework;
+
+namespace Npgsql.Tests.Replication
+{
+    public class PgOutputReplicationTests : SafeReplicationTestBase<NpgsqlLogicalReplicationConnection>
+    {
+        [Test]
+        public Task CreateReplicationSlot()
+            => SafeReplicationTest(nameof(CreateReplicationSlot) + "_pgoutput",
+                async (slotName, _) =>
+                {
+                    await using var c = await OpenConnectionAsync();
+                    TestUtil.MinimumPgVersion(c, "10.0", "The Logical Replication Protocol (via pgoutput plugin) was introduced in PostgreSQL 10");
+                    await using var rc = await OpenReplicationConnectionAsync();
+                    var options = await rc.CreateReplicationSlot(slotName);
+
+                    using var cmd =
+                        new NpgsqlCommand($"SELECT * FROM pg_replication_slots WHERE slot_name = '{options.SlotName}'",
+                            c);
+                    await using var reader = await cmd.ExecuteReaderAsync();
+
+                    Assert.That(reader.Read, Is.True);
+                    Assert.That(reader.GetFieldValue<string>(reader.GetOrdinal("slot_type")), Is.EqualTo("logical"));
+                    Assert.That(reader.GetFieldValue<string>(reader.GetOrdinal("plugin")), Is.EqualTo("pgoutput"));
+                    Assert.That(reader.Read, Is.False);
+                });
+
+        [Test(Description = "Tests whether INSERT commands get replicated as Logical Replication Protocol Messages")]
+        public Task StartReplicationReplicatesInsert()
```

Comment:
Almost all tests are StartReplicationReplicates*. In the interest of brevity, I'd just call this "Insert".

Reply:
👍